### ISSUE TITLE
gradle-plugin corrupts generated code 

### ISSUE THREAD

**Issue by PhilipDukhov** [cqfn/diKTat#569]
I use [sqldelight](https://github.com/cashapp/sqldelight/), it stores generated code inside `module/build/generated/..` folder
Just adding `id("org.cqfn.diktat.diktat-gradle-plugin") version "0.1.5"`, no matter how I configure it with `diktat {}`(module build path not included), adds to the generated code `public` modifiers for all funtions/classes(and probably some other changes), and makes it not buildable.
Here's a [sample project](https://github.com/PhilipDukhov/KMP-server-test/tree/diktatBugExample)

**Comment by petertrr** [comment]
Hi @PhilipDukhov ! Thanks for your feedback.

I don't think diktat changes any visibility modifiers. I've cloned your project and it indeed doesn't build, but diktat tasks (namely `diktatCheck` and `diktatFix`) are not executed before build crashes (I've run `gradlew build`), as far as I can tell. Also removing diktat plugin doesn't fix the build in the branch that you provided. If you have more evidence, please provide your build logs and commands to reproduce.

But you are definitely right, generated code inside `build` directory shouldn't be checked or fixed by diktat. In diktat 0.1.6 we are preparing to add `exclude` property in the plugin, but I'll create a separate issue for automatic exclusion of generated code under the `build` directory.

**Comment by PhilipDukhov** [comment]
My bad, there were other build errors, even when diktat plugin isn't connected, I've updated the repo.
I know it sounds crazy, I spend a couple of hours trying to find out the problem, and was very surprised with the result. 
Following steps works on my side every time:
1. `./gradlew build ` -> `Getter visibility must be the same as property visibility`
2. comment/remove [line 18 of `build.gradle`](https://github.com/PhilipDukhov/KMP-server-test/blob/052be80c57c4c21794f1c5fb9a87507667545e28/build.gradle#L18)
3. `./gradlew build ` -> succeed

**Comment by petertrr** [comment]
After digging in a little bit, I've found the issue. Sqldelight depends on kotlinpoet 1.5.0, while diktat has transitive dependency on kotlinpoet 1.7.1. This is definitely an issue on our side, because we don't need kotlinpoet in runtime. As a workaround, you can declare diktat plugin in buildscript and exclude the dependency. If I change your buildscript in `build.gradle` to
```groovy
buildscript {
    ext.kotlin_version = "1.4.20"
    repositories {
        google()
        jcenter()
    }
    dependencies {
        classpath "com.android.tools.build:gradle:4.0.0"
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
        classpath "com.squareup.sqldelight:gradle-plugin:1.4.4"
        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
        classpath("org.cqfn.diktat:diktat-gradle-plugin:0.1.5") {
            exclude(group: "com.squareup", module: "kotlinpoet")
        }
    }
}
```
and remove `plugins {}` block, I can successfully run `gradlew build`.

**Comment by petertrr** [comment]
Closed in #571
